packages/contract/src/contract-call-script/src/main.sw

```diff
@@ -21,14 +19,33 @@ fn get_script_data_offset() -> u64 {
     is + script_length
 }
 
-fn main() {
-    let script_data_offset = get_script_data_offset();
-    let call_data_offset = script_data_offset + CONTRACT_ID_LEN + WORD_SIZE;
-    let amount_offset = script_data_offset + CONTRACT_ID_LEN;
-    let asset_id_offset = script_data_offset;
+fn get_script_data<T>() -> T {
+    let script_data_ptr = get_script_data_offset();
+    let script_data = asm(r1: script_data_ptr) {
+        r1: T
+    };
+    script_data
+}
+
+struct CallData {
+    contract_id: ContractId,
+    fn_selector: [u8; 8],
```

Comment:
This doesn't look correct. Each `u8` in the array takes up one word. So in total this takes up 8 words.

Reply:
Yeah, just experienced this. Didn't break anything on this PR but when I tried logd-ing based on `size_of<[u8; 8]>()` it was logging too much.

Reply:
Suggest just using a `u64` here, or perhaps a `u32`.